fabric-network/test/transaction.js

```diff
@@ -302,110 +273,33 @@ describe('Transaction', () => {
 			const result = await transaction.evaluate();
 			expect(result).to.equal(expectedResult);
 		});
-
-		it('passes a query to the query handler', async () => {
-			await transaction.evaluate();
-			sinon.assert.calledWith(stubQueryHandler.evaluate, sinon.match.instanceOf(Query));
-		});
-
-		it('builds correct request for no-args invocation', async () => {
-			await transaction.evaluate();
-
-			const query = stubQueryHandler.evaluate.lastArg;
-			expect(query._request).to.deep.include({
-				chaincodeId,
-				fcn: transactionName,
-				txId: transaction.getTransactionID(),
-				args: []
-			});
+		it('returns the bad result from the query handler', async () => {
+			try {
+				queryHandler.evaluate = sinon.stub().resolves(invalidProposalResponse);
+				await transaction.evaluate();
+			} catch (error) {
+				expect(error.status).to.equal(500);
+			}
 		});
+	});
 
-		it('builds correct request for with-args invocation', async () => {
+	describe('#_buildRequest', () => {
```

Comment:
Unit tests should be aiming to drive code through the use of the public API. Testing internal implementation functions directly is less useful for ensuring end-to-end code correctness.